### PR TITLE
[SC-251] Initial commit for docker AMI

### PR DESCRIPTION
**Pull request by zaro0508**
Use packer to Create an AMI with docker and a few tools.  This
AMI will be used for the new SC docker product.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,36 +3,40 @@ language: python
...
+    if: tag =~ /^v([0-9]+)\.([0-9]+)\.([0-9]+)(?:([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+[0-9A-Za-z-]+)?$/
 jobs:
   include:
     - stage: validate
       script:
         - pre-commit run --all-files
         - pushd src
         - echo "ImageName = ${IMAGE_NAME}"
-        - packer validate -var ImageName=${IMAGE_NAME} template.json
+        - packer validate -var ImageName=${IMAGE_NAME} -var SourceImage="some_aws_image_path" template.json
```
Comment (tthyer) [inline .travis.yml:32]: Why is this addition necessary?

File: .travis.yml [hunk 2]
```diff
@@ -3,36 +3,40 @@ language: python
...
     - stage: deploy-branch
       script:
+        - export LATEST_AMI=$(aws ssm get-parameters --profile ${AWS_PROFILE} --names ${AWS_SOURCE_IMAGE_PATH} --query 'Parameters[0].[Value]' --output text)
         - pushd src
-        - travis-wait-improved --timeout 30m packer build -force -var PACKER_LOG=1 -var AwsProfile=${AWS_PROFILE} -var AwsRegion=${AWS_REGION} -var ImageName=${IMAGE_NAME} template.json
+        - travis-wait-improved --timeout 30m packer build -force -var PACKER_LOG=1 -var AwsProfile=${AWS_PROFILE} -var ImageName=${IMAGE_NAME} -var SourceImage=${LATEST_AMI} template.json
```
Comment (tthyer) [inline .travis.yml:37]: So we don't have to specify region anymore for this?

File: README.md [hunk 3]
```diff
@@ -1,9 +1,6 @@
 # packer-ami-template
```
Comment (tthyer) [inline README.md]: Change this header please

File: README.md [hunk 4]
```diff
@@ -1,9 +1,6 @@
...
 A template for quickly getting a new packer AWS AMI project started.
```
Comment (tthyer) [inline README.md]: Use an appropriate description, please

File: src/playbook.yaml [hunk 5]
```diff
@@ -1,12 +1,49 @@
...
 
   tasks:
-    - name: install system packages
-      shell: "yes | aptdcon --hide-terminal --install 'build-essential'"
+    - name: Upgrade all packages
+      yum: name=* state=latest
+
+    - name: Install system packages
+      ansible.builtin.yum:
+        name: "{{ item }}"
```
Comment (tthyer) [inline src/playbook.yaml:13]: why is this in each step: `name: "{{ item }}"`?

File: src/playbook.yaml [hunk 6]
```diff
@@ -1,12 +1,49 @@
...
+        state: present
+      with_items:
+        - "curl"
+        - "wget"
+        - "jq"
+        - "git"
+        - "amazon-linux-extras"
+        - "docker"
```
Comment (tthyer) [inline src/playbook.yaml:21]: nice to not have to add the gpg key

File: src/playbook.yaml [hunk 7]
```diff
@@ -1,12 +1,49 @@
...
+
+    - name: Start docker service
+      ansible.builtin.service:
+        name: docker
+        state: started
```
Comment (tthyer) [inline src/playbook.yaml:26]: why must you start it during the build? what does it do/make?

File: src/playbook.yaml [hunk 8]
```diff
@@ -1,12 +1,49 @@
...
+
+    - name: Add the ec2-user to the docker group
+      ansible.builtin.user:
+        name: "{{ item }}"
+        groups: docker
+        append: yes
+      with_items:
+        - ec2-user
```
Comment (tthyer) [inline src/playbook.yaml:34]: Nice ansible tool to do this

File: src/playbook.yaml [hunk 9]
```diff
@@ -1,12 +1,49 @@
...
+
+    - name: Make docker auto start
+      ansible.builtin.service:
+        name: "{{ item }}"
+        enabled: yes
+      with_items:
+        - docker
```
Comment (tthyer) [inline src/playbook.yaml:41]: Another nice ansible tool

File: src/playbook.yaml [hunk 10]
```diff
@@ -1,12 +1,49 @@
...
 
     - name: install python packages
-      pip:
-        name:
-          - synapseclient
+      ansible.builtin.pip:
```
Comment (tthyer) [inline src/playbook.yaml:44]: so many cool builtins

**Review by tthyer — APPROVED** [review]
@zaro0508 Looks very nice, I just have a few questions

File: .travis.yml (hunk 1 above)
Comment (zaro0508) [reply .travis.yml:32]: i updated the packer file to take a `SourceImage` as input so now it's a required parameter.  for validating we just need to pass in a dummy value.

File: .travis.yml (hunk 2 above)
Comment (zaro0508) [reply .travis.yml:37]: correct, we assume region to be set in the AWS profile config so we only need to pass in profile.

File: src/playbook.yaml (hunk 5 above)
Comment (zaro0508) [reply src/playbook.yaml:13]: that's what ansible recommends or maybe I followed a pattern established by some yahoo on the internet?

File: src/playbook.yaml (hunk 7 above)
Comment (zaro0508) [reply src/playbook.yaml:26]: not sure but i think i was just following this yahoo's instructions.. https://github.com/Sage-Bionetworks-IT/packer-amazonlinux-docker/pull/1/files#diff-606e9b44b66f176373a6e9b820e9faf2a705817c3ac893deca167f7abee58b13R2

File: README.md (hunk 3 above)
Comment (zaro0508) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (zaro0508) [reply README.md]: done
